graphql-server/pkg/resolvers/proposal.go

```diff
@@ -87,6 +87,100 @@ func (r *proposalResolver) MyReaction(ctx context.Context, obj *models.Proposal)
 	return &reaction.Reaction, nil
 }
 
+func (r *proposalResolver) Votes(ctx context.Context, obj *models.Proposal, input models.QueryProposalVotesInput) (*models.Connection[models.ProposalVote], error) {
+	result := make([]models.ProposalVote, 0)
+
+	validatorLimit := input.First
+	validatorOffset := input.After
+	// No need validators if no delegation or already went through all the validators
+	if len(input.PinnedValidators) == 0 || input.After > len(input.PinnedValidators) {
+		validatorLimit = 0
+		validatorOffset = 0
+	}
+
+	validators, err := pkgContext.GetQueriesFromCtx(ctx).Validator.WithProposalVotes([]int{obj.ID}).QueryPaginatedValidators(validatorLimit, validatorOffset, input.PinnedValidators)
+	if err != nil {
+		return nil, servererrors.QueryError.NewError(ctx, fmt.Sprintf("failed to load validators: %v", err))
+	}
+
+	// Add validators to result
+	validatorDelegationAddresses := make([]string, 0)
+	for i := range validators.Items {
+		validator := validators.Items[i]
+		if validator.Info != nil && len(validator.Info.ProposalVotes) != 0 {
+			result = append(result, *validator.Info.ProposalVotes[0])
```

Comment:
why `[0]`? if a validator voted multiple times will the query return latest vote?

Reply:
the vote gets overwritten when voted multiple times
https://github.com/forbole/bdjuno/blob/50e97941f2ae69dfba6e2382c6c16cd641464fd4/database/gov.go#L270-L275

`ProposalVotes` itself has a many-to-one relationship with validators but when the proposal id scope is applied it becomes one-to-one so `len(validator.Info.ProposalVotes)` would always be either 1 or 0 here

maybe think think if there are better ways to handle this

Reply:
Maybe I change the proposal id scope to singular first to ensure there's only 1 vote